doc/user/methods/random_forest.md

```diff
@@ -712,3 +717,23 @@ class CustomCategoricalSplit
    dataset will be used to train each decision tree.
  * If `false` _(default for the `ExtraTrees` [variant](#fully-custom-behavior))_, the full
    dataset will be used to train each decision tree.
+ * This parameter will be removed in future versions. A value of `false` will then be
+   equivalent to `BootstrapType` `IdentityBootstrap` and a value of `true` will be
+   equivalent to `DefaultBootstrap`.
+
+#### `BootstrapType`
+
+Classes compatible with the `BootstrapType` concept support a single bool class template argument `UseWeights` to specify
+whether classes `Bootstrap()` method should bootstrap the weights or not.
+
+The concept requires a method with the following signature:
+
+`void Bootstrap(dataset, labels, weights, bootstrapDataset, bootstrapLabels, bootstrapWeights);`
+
+The library provides three implementations for the `BootstrapType` concept.
+
+| **name** | **description** |
+|----------|-----------------|
+| `IdentityBootstrap` | This implementation is a no-op. It just copies the input `dataset`, `labels`, and `weights` to the output `bootstrapDataset`, `bootstrapLabels`, and `bootstrapWeihts` |
+| `DefaultBootstrap` | This implementation of a bootstrap is a random sampling with replacement. |
+| `SequentialBootstrap` | This implementation of a bootstrap is a random sampling with replacement as presented in: M. López de Prado (2018): "Advances in Financial Machine Learning", pp. 63-65 |
```

Comment:
I know this is a little pedantic but I think it's really important that documentation styling is unified.  Primarily I just changed the formatting to match other template parameters, and added documentation for parameters to the `Bootstrap()` method.  I also made minor wording changes to try and make things a little shorter / more consistent with other sections.

Don't feel obligated to take the suggestion as-is, but I would like if we could make the way this subsection looks match the other subsections about the other template parameters.

```suggestion

 * Specifies the strategy used for bootstrapping data for each .
 * Three implementations for `BootstrapType` are available for drop-in usage:
   - `DefaultBootstrap` *(default)*: bootstrap via random sampling with replacement.
   - `IdentityBootstrap`: no bootstrapping.  Simply copies the input `dataset`, `labels`, and `weights` for each tree's data.
   - `SequentialBootstrap`: bootstrapping via random sampling with replacement as presented in: M. López de Prado (2018): "Advances in Financial Machine Learning", pp. 63-65.
 * A custom `BootstrapType` class must take a `bool` template parameter `UseWeights` and implement one function:

```c++
template<bool UseWeights>
class CustomBootstrapType
{
  /**
   * Compute a bootstrap dataset based on the original dataset.
   * If `UseWeights` is `false`, then `weights` and `bootstrapWeights` can be
   * ignored.
   *
   * When the function is complete, `bootstrapDataset` and `bootstrapLabels`
   * should contain a bootstrapped dataset.  If `UseWeights` is `true`, then
   * `bootstrapWeights` should contain the corresponding instance weights for
   * the bootstrapped dataset.
   */
  template<typename MatType,
           typename LabelsType,
           typename WeightsType>
  void Bootstrap(const MatType& dataset,
                 const LabelsType& labels,
                 const WeightsType& weights,
                 MatType& bootstrapDataset,
                 LabelsType& bootstrapLabels,
                 WeightsType& bootstrapWeights)
};
```

Also:

 * There's no easily available PDF of the `Advances in Financial Machine Learning` book, but we tend to try and link to papers wherever possible.  Do you think we could link to the other paper you recommended?  The PDF isn't directly and freely available, but [it's at least got its own page on Elsevier](https://www.sciencedirect.com/science/article/abs/pii/S0378375897000414) so people at institutions should be able to get a PDF.  Maybe we can reference both, I'm fine with that.
 
 * I'm not deeply familiar with the technique you linked to, but do you think that we could expand beyond describing `SequentialBootstrap` as `random sampling with replacement`?  That doesn't differentiate from `DefaultBootstrap`.  My understanding is that `SequentialBootstrap` is weighted random sampling with replacement such that the final bootstrap sample is equally informative (for some definition of 'informative'... but I can only see the abstract :)).

Reply:
I updated the documentation. I checked the author's webpage (https://quantresearch.org/Publications.htm), but there is no document available. I think referencing the other document is misleading.